src/DependencyInjection/Compiler/AddPsyshCommandPass.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the PsyshBundle package.
+ *
+ * (c) Théo FIDRY <theo.fidry@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Fidry\PsyshBundle\DependencyInjection\Compiler;
+
+use Psy\Command\Command as PsyshCommand;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Compiler pass allowing to add Psysh commands dynamically
+ *
+ * @author Jérôme Vieilledent <jerome@vieilledent.fr>
+ */
+class AddPsyshCommandPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->has('psysh.shell')) {
+            return;
+        }
+
+        // Workaround to avoid Psysh commands to be registered as regular console commands
+        // (conflict with service autoconfiguration as Psysh commands inherit from \Symfony\Component\Console\Command\Command as well
+        // Note that this compiler pass must run with a higher priority than AddConsoleCommandPass to be efficient.
+        foreach ($container->findTaggedServiceIds('console.command') as $id => $attributes) {
+            $definition = $container->findDefinition($id);
+            if (is_subclass_of($definition->getClass(), PsyshCommand::class)) {
+                $definition->clearTag('console.command');
+            }
+        }
+
+        $commands = [];
+        foreach ($container->findTaggedServiceIds('psysh.command') as $id => $attributes) {
```

Comment:
instead of the previous loop, cant you just remove the console.command tag in this one?